charts/tekton-apps/templates/triggers.yaml

```diff
@@ -0,0 +1,83 @@
+{{- range $project := .Values.apps }}
+{{- if $project.enabled }}
+{{- range $component := $project.components }}
+
+{{/* initialize `projectEnvironment` variable with either `$project.environment` if it exists or with `$.Values.environment` otherwise */}}
+{{ $projectEnvironment := ternary $project.environment $.Values.environment (hasKey $project "environment") -}}
```

Comment:
Can you use `coalesce` here instead of `ternary` and in other places? Will it work? https://helm.sh/docs/chart_template_guide/function_list/#coalesce

Reply:
done